src/test/documentation_examples.rs

```diff
@@ -1414,9 +1422,74 @@ async fn versioned_api_examples() -> Result<()> {
     let client = Client::with_options(options)?;
     // End Versioned API Example 4
 
+    let mut options = ClientOptions::parse(&uri).await?;
+    let server_api = ServerApi::builder()
+        .version(ServerApiVersion::V1)
+        .strict(true)
+        .build();
+    options.server_api = Some(server_api);
+    let client = Client::with_options(options)?;
+    let db = client.database("versioned-api-migration-examples");
+    db.collection::<Document>("sales").drop(None).await?;
+    // Start Versioned API Example 5
+    db.collection("sales").insert_many(vec![
+        doc! { "_id" : 1, "item" : "abc", "price" : 10, "quantity" : 2, "date" : iso_date("2021-01-01T08:00:00Z")? },
```

Comment:
ideally this would be copy-pastable for users, so I think we want leverage the `chrono` interop here and do:
```rust
chrono::DateTime::parse_from_rfc3339("...")?
```
Since this is feature gated, we'll need to add a comment at the top of the example indicating that users need to enable the `bson-chrono-0_4` feature flag.

You might have to do some hackery to get the `?` operator to work, maybe we could update this function to return `Result<(), Box<dyn Error>>` or something to ease that.

Reply:
`bson-chrono-0_4` isn't enabled by default*, so doing it this way required adding `bson` with that feature as a `dev-dependency` (and also updating to `resolver = "2"` so the features didn't get merged into the regular build).  It's a little unfortunate to have to update the `bson` version in two places; the other option would be to keep the conversion function (which can be made much shorter thanks to `chrono::DateTime::parse_from_rfc3339`) and pull it into the example with a note that it wouldn't be needed if `bson-chrono-0_4` was enabled.

\*  Side note: why not? `chrono`'s pulled in either way.

Reply:
Correction: `resolver = "2"` isn't in our MSRV, so I switched to the other option.

Reply:
Ah yeah that is a bit tricky. I think your solution of including the conversion function in the example is a good choice for now. It made me realize that we probably want to have a `parse_from_rfc3339` method on `bson::DateTime` even without the feature flag, so I filed RUST-977 for that.

>Side note: why not? chrono's pulled in either way.

The feature flag is to protect us from instability in `chrono`'s API rather than from it as a dependency. See https://github.com/mongodb/bson-rust/pull/260#pullrequestreview-676430020 and the [C-STABLE](https://rust-lang.github.io/api-guidelines/necessities.html#c-stable) Rust API guideline for more detail.